_book/04-advanced-topics/04-controlling-the-svg-output.md

```diff
@@ -1,11 +1,128 @@
+      options:
+        breaks: 'auto'
+        spacingStaff: 12
+        adjustPageHeight: False
+        header: 'auto'
+        footer: 'auto'
+        justifyVertically: True
+
+    - name: 02-units
+      test-suite: score/score-013.mei
+      options:
+        breaks: 'auto'
+        unit: 6.0
+        spacingStaff: 12
+        adjustPageHeight: False
+        header: 'auto'
+        footer: 'auto'
+        justifyVertically: True
+
+    - name: 03-page-size
+      test-suite: score/score-013.mei
+      options:
+        pageHeight: 3050
+        pageWidth: 2290
+        breaks: 'auto'
+        scale: 41
+        spacingStaff: 12
+        adjustPageHeight: False
+        header: 'auto'
+        footer: 'auto'
+        justifyVertically: True
+
+    - name: 04-responsive-page
+      test-suite: score/score-013.mei
+      options:
+        adjustPageHeight: False
+        breaks: 'auto'
+        pageHeight: 800
+        pageWidth: 1800
+        scale: 100
+        scaleToPageSize: True
+        spacingStaff: 12
+
+    - name: 05-responsive-page
+      test-suite: score/score-013.mei
+      options:
+        adjustPageHeight: False
+        breaks: 'auto'
+        pageHeight: 800
+        pageWidth: 1800
+        scale: 30
+        scaleToPageSize: True
+        spacingStaff: 12
 ---
 
-[in preparation]
+### Page units and dimensions
+
+Verovio uses an abstract unit for specifying the page dimensions. By default, the page height is `2970` and the page width is `2100`. These are equivalent to the dimension of an A4 page in tenths of a millimeter. When generating SVG, these units are interpreted as pixels, which means that the default SVG image size is **2970px** height by **2100px** width. 
+
+The example below shows an empty page with the default dimensions – and the option `--justifyVertically` turned on.
+
+{% include music-notation-only example="01-default" %}
+
+The option `--mm-output` can be turned on to have the page dimension being interpreted as millimeters. In that case, the SVG image will have a size of **297mm** by **210mm**, which is recommended if the SVG is meant to be printed or converted to PDF.
+
+Page margins (`--page-margin-bottom`,  `--page-margin-left`,  `--page-margin-right` and  `--page-margin-top`) are all specified in abstract units, with a default value of `50`. That is **50px** with the default SVG output and **5mm** with the `--mm-output` option turned on.
+
+#### MEI unit
+
+Most of the options in Verovio are given in MEI units. An MEI unit (or MEI virtual unit) corresponds to half the distance between adjacent staff lines where the interline space is measured from the middle of a staff line. The value of the MEI unit in Verovio will determine the size of the staff on a page. By default, the MEI unit is `9.0`, which means that a staff space is 9 abstract units. In terms of staff size (or raster), it means a staff size of **72px** in the default SVG output, or **7.2mm** with the `--mm-output` option turned on.
```

Comment:
```suggestion
Many of the measurements in Verovio are given in MEI units. An MEI unit (or MEI virtual unit) is an abstract unit; that is, it does not correspond to any specific unit such as pixels or millimetres. An MEI unit corresponds to half the distance between adjacent staff lines where the interline space is measured from the middle of a staff line. The value of the MEI unit in Verovio will determine the size of the staff on a page. By default, the MEI unit is `9.0`, which means that a staff space is 9 abstract units. In terms of staff size (or raster), the default unit value corresponds to a staff size of **72px** in the default SVG output (8px/unit), or **7.2mm** (0.8mm/unit) with the `--mm-output` option enabled.
```

Reply:
I don't understand `(8px/unit)`. What does it mean?

Reply:
If the default unit is "9" and the default staff height is "72px" then that's 72px / 9u = 8px/u?

Reply:
OK -- so a unit value of 15 would be 15px/unit. 

I got confused with "staff space is 9 abstract units" -- this isn't really correct then? A staff space is 2 "abstract units", or 18px, or 1.8mm. 

A full "staff" (five lines) is 8u / 72px / 7.2mm.

Reply:
Is that the measurement to the middle of each staff line, or are the staff lines added to the total height? So a staff might be 8u, and the staff lines are 2px high, then is it 72px total, or 72px + (2px * 5) = 82px?

Reply:
The measurement is to the middle of the staff line. So the total staff height would be 74px

Reply:
I'm wondering if it's better to leave out the "72px" then, since the total height would actually be `(unit * 8) + (((staffLineWidth * unit) * 2) / 2)`? (e.g., `(9 * 8) + (((0.15 * 9) * 2) / 2)` by default)

Reply:
I wasn't suggesting that we remove the table, though! I agree that it is useful.

Reply:
I was calculating with the 2px you gave as example above.

We can say "72px (excluding staff line width)", which seems logical since it is stated that spaces are calculated from the middle of the line. 

This is a standard practice I think. You can look in Behind Bars, or for the Dorico documentation that is very similar https://steinberg.help/dorico/v1/en/dorico/topics/notation_reference/notation_reference_staves_staff_size_changing_layouts_t.html

Reply:
Yeah, I gave my first example before looking up how `staffLineWidth` was defined... sorry for the confusion.

I think it's fine to explain it, but was just trying to figure out how to reconcile the documentation with the user's experience. If a user measures their "default" staff rendering and finds it to be more than 72px then they might question whether the documentation needs to be corrected, since the value of 72px doesn't really exist anywhere in the output (I think). 

I'm even questioning if we need to mention 72px at all, since we've just said that the MEI virtual units, and Verovio abstract units, are not tied to a "physical" unit size at all. So it seems a bit disjointed to contradict that in the last sentence of the paragraph.

Reply:
Oh, I was thinking more in terms of CSS control, positioning, etc. It's not a big deal -- we can keep it as 72 if you want.